addons/mail/static/src/models/device/device.js

```diff
@@ -59,6 +60,23 @@ function factory(dependencies) {
         globalWindowInnerHeight: attr(),
         globalWindowInnerWidth: attr(),
         isMobile: attr(),
+        /**
+         * Size class of the device.
+         *
+         * This is an integer representation of the size.
+         * Useful for conditional based on a device size, including
+         * lower/higher. Device size classes are defined in sizeClasses
+         * attribute.
+         */
+        sizeClass: attr(),
+        /**
+         * Size classes per device size.
+         *
+         * Useful to make conditional based on current device size (class).
+         */
+        sizeClasses: attr({
+            default: { XS: 0, VSM: 1, SM: 2, MD: 3, LG: 4, XL: 5, XXL: 6 },
+        }),
```

Comment:
Why make this a field? They aren't record specific and shouldn't have to change dynamically. I think it's fine to read these from `env.device.SIZES` directly.

Reply:
`env.device.SIZES` was fine in prod code, but it crashes in test if we don't setup `SIZES` during `env` creation of the specified tests. IMO this was stupid, so I just prefer to not deal with mocking data in env and always define those size classes, regardless of how `env.device` is setup.

Yes, it's dup with `env.device`. `env.messaging.device` embraces a better architecture, especially for testing and model integration, so I tend to prefer this over `env.device`.

Reply:
It should be setup in `makeTestEnvironment` in web with other device keys, it's just missing.

Reply:
I made the changes. I still think the mocking of `env.devices` is stupid, but fine...